typed-racket-lib/typed-racket/base-env/base-env.rkt

```diff
@@ -1179,8 +1179,8 @@
 [procedure? (make-pred-ty top-func)]
 [compose (-poly (a b c) (-> (-> b c) (-> a b) (-> a c)))]
 [compose1 (-poly (a b c) (-> (-> b c) (-> a b) (-> a c)))]
-[procedure-rename (-> top-func -Symbol top-func)]
-[procedure->method (-> top-func top-func)]
+[procedure-rename (-poly (a) (-> (-Inter top-func a) -Symbol (-Inter top-func a)))]
```

Comment:
Why do we need the intersection on the result type?

Reply:
Actually, we don't. I'm pushing a fix on that.